app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -127,6 +127,10 @@ def user_data_supported?
       true
     end
 
+    def zones
+      @zones ||= client.list_zones.data[:body]["availabilityZoneInfo"].try(:map){|i| i["zoneName"]}
```

Comment:
a) use .body rather than .data[:body] for consistency
b) using try here may cause a return of nil, which will cause a UI error